homeassistant/components/filter/sensor.py

```diff
@@ -214,6 +219,9 @@ def _update_filter_sensor_state(self, new_state, update_ha=True):
         if self._icon is None:
             self._icon = new_state.attributes.get(ATTR_ICON, ICON)
 
+        if self._device_class is None:
+            self._device_class = new_state.attributes.get(ATTR_DEVICE_CLASS)
```

Comment:
We're not limiting the source entity_id to the sensor domain so we need to validate that the device class of the source entity is a sensor device class.

https://github.com/home-assistant/core/blob/5b14c1da7314a6b84cb1e5475169967d99902c46/homeassistant/components/sensor/__init__.py#L37

Reply:
Actually it would make sense to limit it to the sensor domain. How would one accomplish such ? Any idea idea of an integration that already does that ?

Reply:
We can use `cv.entity_domain` as an additional validator in the platform schema for the entity_id item.

Reply:
Just realised the throttle filter can be used for binary_sensors too... 

But can we have been talking about domain when we mean platform right ?

Reply:
The entity domain is the word before the period in the entity_id.

Reply:
Right!

https://github.com/home-assistant/core/blob/ad634a393b352b8f8b3000debad386cef0cce5b4/homeassistant/components/filter/__init__.py#L3

Confused me there

Reply:
So now we need this device class validation check here, right?

Reply:
do we 😕 ? (sorry I'm confused here)

It's either `sensor` or `binary sensor `(both domains have `device_class`)

Reply:
We're setting the source entity device class to this filter sensor device class. Only sensor entity device classes may be used for sensor platform entities.

Reply:
thanks for clarifying!